textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/regexmatch/RegexMatcherTest.java

```diff
@@ -26,46 +26,141 @@
  * Created by chenli on 3/25/16.
  */
 public class RegexMatcherTest {
-    
-    private RegexMatcher regexMatcher;
-    private IDataWriter dataWriter;
-    private IDataReader dataReader;
-    private IDataStore dataStore;
-    
-    @Before
-    public void setUp() throws Exception{
-        dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR, TestConstants.SAMPLE_SCHEMA_PEOPLE);
-        dataWriter = new LuceneDataWriter(dataStore);
-        dataWriter.clearData();
-        dataWriter.writeData(TestConstants.getSamplePeopleTuples());
-        dataReader = new LuceneDataReader(dataStore,LuceneConstants.SCAN_QUERY, 
-            TestConstants.SAMPLE_SCHEMA_PEOPLE.get(0).getFieldName());
-    }
-    
-    @After
-    public void cleanUp() throws Exception{
-        dataWriter.clearData();
-    }
-    
-    @Test
-    public void testGetNextTuple() throws Exception{
-        String regex = "b.*"; //matches bruce and brad
-        String fieldName = TestConstants.FIRST_NAME;
-        IPredicate predicate = new RegexPredicate(regex, fieldName);
-        ISourceOperator sourceOperator = new ScanBasedSourceOperator(dataReader);
-        List<ITuple> tuples = TestConstants.getSamplePeopleTuples();
-        
-        regexMatcher = new RegexMatcher(predicate, sourceOperator);
-        regexMatcher.open();
-        ITuple nextTuple = null;
-        int numTuples = 0;
-        while((nextTuple = regexMatcher.getNextTuple()) != null){
-            boolean contains = TestUtils.contains(tuples, nextTuple);
-            Assert.assertTrue(contains);
-            numTuples ++;
-        }
-        Assert.assertEquals(2, numTuples);
-        regexMatcher.close();
-    }
-
-}
\ No newline at end of file
+
+	private RegexMatcher regexMatcher;
+	private IDataWriter dataWriter;
+	private IDataReader dataReader;
+	private IDataStore dataStore;
+
+	public void setUpPeople() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR,
+				TestConstants.SAMPLE_SCHEMA_PEOPLE);
+		dataWriter = new LuceneDataWriter(dataStore);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSamplePeopleTuples());
+	}
+
+	public void setUpCorp() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR,
+				TestConstants.SAMPLE_SCHEMA_CORP);
+		dataWriter = new LuceneDataWriter(dataStore);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSampleCorpTuples());
+	}
+
+	public void setUpStaff() throws Exception {
+		dataStore = new LuceneDataStore(LuceneConstants.INDEX_DIR,
+				TestConstants.SAMPLE_SCHEMA_STAFF);
+		dataWriter = new LuceneDataWriter(dataStore);
+		dataWriter.clearData();
+		dataWriter.writeData(TestConstants.getSampleStaffTuples());
+
+	}
+
+	@After
+	public void cleanUp() throws Exception {
```

Comment:
Since we have different "setUp" functions for different domains, why do we have a single "cleanUp" function?  Just curious.  I guess it's related to how "test" works?